content/en/blog/_posts/2024-04-22-userns-beta/index.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: blog
+title: "Kubernetes 1.30: User Namespaces hits Beta!"
+date: 2024-04-22
+slug: userns-beta
+---
+
+**Authors:** Rodrigo Campos Catelin (Microsoft), Giuseppe Scrivano (Red Hat), Sascha Grunert (Red Hat)
+
+Linux provides different namespaces to isolate processes from each other. For
+example, a typical Kubernetes pod runs within a network namespace to isolate the
+network identity, a PID namespace to isolate the processes.
+
+One Linux namespace that was left behind is the [user
+namespace](https://man7.org/linux/man-pages/man7/user_namespaces.7.html). This
+namespace allows us to isolate the user and group identifiers (UIDs and GIDs) we
+use inside the container from the ones on the host.
+
+This is a powerful abstraction that allows us to run containers as "root": we
+are root inside the container and we can do everything root can inside the pod,
+but our interactions with the host are limited to what a non-privileged user can
+do. This is great to limit the impact of a container breakout.
+
+A container breakout is when a process inside a container is able to break out
+onto the host using some unpatched vulnerability in the container runtime or the
+kernel and able to access / modify files on the host or other containers. If we
+run our pods with user namespaces, the privileges the container has over the
+rest of the host are reduced and the files outside the container it can access
+are limited too.
+
+In Kubernetes v1.25 we introduced support for user namespaces for only stateless
+pods. Kubernetes 1.28 lifted that restriction and now, with Kubernetes 1.30, we
+are moving to beta!
+
+## What is a user namespace?
+
+Note: Linux user namespaces are a different concept from [kubernetes
+namespaces](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/).
```

Comment:
We use site-relative links where feasible.

```suggestion
Note: Linux user namespaces are a different concept from [kubernetes
namespaces](/docs/concepts/overview/working-with-objects/namespaces/).
```

Reply:
Fixed, thanks!